main.nf

```diff
@@ -615,145 +597,250 @@ process calculateNSCRSC {
 
 
 /*
- * STEP 7 deepTools
+ * STEP 7.1 deepTools bamPEFragmentSize
  */
+bam_dedup_deepTools.into {
+    bam_dedup_deepTools_bamPEFragmentSize;
+    bam_dedup_deepTools_plotFingerprint1;
+    bam_dedup_deepTools_plotFingerprint2;
+    bam_dedup_deepTools_bamCoverage;
+    bam_dedup_deepTools_multiBamSummary1;
+    bam_dedup_deepTools_multiBamSummary2
+}
+
+process deepTools_bamPEFragmentSize {
+    tag "$bam_base"
+    publishDir "${params.outdir}/deepTools/FragmentSize", mode: 'copy'
+    label 'process_big'
+
+    input:
+    set file(bam), file(bai) from bam_dedup_deepTools_bamPEFragmentSize
+
+    output:
+    file '*.{txt,pdf}' into deepTools_bamPEFragmentSize_results
+    file '*.txt' into deepTools_bamPEFragmentSize_multiqc
+
+    when:
+    !params.singleEnd
+
+    script:
+    bam_base = bam.baseName - '.dedup.sorted'
+    """
+    bamPEFragmentSize \\
+        --binSize 1000 \\
+        --bamfiles $bam \\
+        --o ${bam_base}.fragment_size_histogram.pdf \\
+        --numberOfProcessors ${task.cpus} \\
+        --plotFileFormat pdf \\
+        --plotTitle "${bam_base}: Paired-end Fragment Size Distribution" \\
+        --outRawFragmentLengths ${bam_base}.bamPEFragmentSize_rawdata.txt
+    """
+}
 
-process deepTools {
-    tag "${bam[0].baseName}"
+/*
+ * STEP 7.2 deepTools plotFingerprint
+ */
+process deepTools_plotFingerprint {
     publishDir "${params.outdir}/deepTools", mode: 'copy'
+    label 'process_big'
 
     input:
-    file bam from bam_dedup_deepTools.collect()
-    file bai from bai_dedup_deepTools.collect()
-    file bed from bed
+    file(bam) from bam_dedup_deepTools_plotFingerprint1.flatten().filter{ !it.toString().contains(".bam.bai") }.collect()
+    file(bai) from bam_dedup_deepTools_plotFingerprint2.flatten().filter{ it.toString().contains(".bam.bai") }.collect()
```

Comment:
There has _got_ to be a better way to do this..

@pditommaso - is there a nicer syntax for this? I want to write this, but it doesn't work:
```nextflow
input:
file(bam), file(bai) from bam_dedup_deepTools_plotFingerprint.collect()
```

I feel like I'm missing something obvious but it's late and I'm going to bed 😅

Reply:
It would be nice to have a magic way to split collection into a tuple of collection. However the best trick I can see here is just keep 

```
file(bam) from bam_dedup_etc.collect()
```

Then in the script use Bash expansion to get only the bam files!

```
-bamfiles *.bam
```

Reply:
Glob expansion, of course 🤦‍♂️ 

Thank you! Sounds like I'm not missing something obvious at least, which is what I was mostly worried about 😆